src/test/java/com/example/statussvc/controller/HostsControllerTest.java

```diff
@@ -0,0 +1,182 @@
+            WHEN performing POST request
+            THEN return response with code 201, valid location and empty body
+            """)
+    void postCorrectData() throws Exception {
+        // GIVEN
+        final String EXPECTED_RESPONSE_URL = "/api/v1/hosts/100";
+        HostCreateRequest hostCreateRequest = HostCreateRequest.builder()
+                .title("Google")
+                .description("Google Description")
+                .url("https://google.com/")
+                .build();
+        given(hostsService.create(hostCreateRequest)).willReturn(100L);
+        // WHEN
+        MockHttpServletResponse actualResponse = mockMvc
+                .perform(post(URL)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(hostCreateRequest))
+                )
+                // THEN
+                .andExpect(status().isCreated())
+                .andReturn()
+                .getResponse();
+        // AND THEN
+        assertThat(actualResponse.getHeader(HttpHeaders.LOCATION))
+                .isNotEmpty()
+                .isEqualTo(EXPECTED_RESPONSE_URL);
+        assertThat(actualResponse.getContentAsString()).isEmpty();
+    }
+
+    @Test
+    @DisplayName("""
+            GIVEN invalid hostCreateRequest object
+            WHEN performing POST request
+            THEN return response with code 400 and message "Bad Request"
+            """)
+    void postInvalidData() throws Exception {
+        // GIVEN
+        HostCreateRequest hostCreateRequest = HostCreateRequest.builder()
+                .title("Google")
+                .description("Google Description")
+                .url(null)
+                .build();
+        given(hostsService.create(hostCreateRequest)).willReturn(101L);
+        // WHEN
+        mockMvc
+                .perform(post(URL)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(hostCreateRequest))
+                )
+                // THEN
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    @DisplayName("""
+            GIVEN default page number and page size
+            WHEN performing GET request
+            THEN return response with code 200 and list of hosts
+            """)
+    void getAllCorrectData() throws Exception {
+        // GIVEN
+        final String EXPECTED_RESPONSE = "{\"totalItems\":2,\"hosts\":[{\"id\":1,\"title\":\"Yahoo\",\"description\":null,\"url\":\"http://yahoo.com/\",\"connectionTime\":null,\"lastCheck\":null,\"status\":\"UNKNOWN\"},{\"id\":2,\"title\":\"Google\",\"description\":\"Google Engine\",\"url\":\"https://google.com/\",\"connectionTime\":null,\"lastCheck\":null,\"status\":\"UNKNOWN\"}],\"totalPages\":1,\"currentPage\":0}";
```

Comment:
eate object and use Object Mapper to verify/transform request/response. that's. not a good practice to hardcode JSON string